src/mlpack/bindings/python/mlpack/matrix_utils.py

```diff
@@ -52,10 +52,19 @@ def to_matrix(x, dtype=np.double, copy=False):
       return x.copy("C"), True
     else:
       return x, False
+  elif (isinstance(x, np.ndarray) and x.dtype == dtype and x.flags.f_contiguous):
+    if copy: # Copy the matrix if required.
+      return np.ndarray(x.shape,buffer=x.flatten(),dtype=dtype,order = 'C').copy("C"),True
```

Comment:
Doesn't this implicitly transpose the matrix?  I think that we should test this by making another test for `test_python_binding.py` where we pass a numpy matrix with `F_CONTIGUOUS` set.  (You can just make a random matrix with `order='F'`.)

Reply:
No I think it doesn't transpose the matrix?
```
df = pd.read_csv('rann_test_q_3_100.csv',header=None)
a = pd.Series(np.random.rand(100))
print(df.values.flags)
```
it gives
```
  C_CONTIGUOUS : False
  F_CONTIGUOUS : True
  OWNDATA : False
  WRITEABLE : True
  ALIGNED : True
  WRITEBACKIFCOPY : False
  UPDATEIFCOPY : False
```
when I compare `df.values` and `to_matrix(df.values)[0]`
as `np.array_equal(df.values,to_matrix(df.values)[0])` it results `True`

Reply:
Huh, this is very strange to me.  Take a look at this Python session:

```
>>> import numpy as np
>>> x = np.random.rand(5, 5)
>>> x = np.array(x, order='F', copy=True)
>>> x
array([[0.4374804 , 0.30758627, 0.97045243, 0.35728207, 0.95584896],
       [0.41457004, 0.82433351, 0.44260369, 0.80797501, 0.78434397],
       [0.10598124, 0.98438937, 0.13709334, 0.24600426, 0.20042625],
       [0.73949954, 0.54399577, 0.64043765, 0.52684565, 0.89393217],
       [0.61319107, 0.67340923, 0.37922999, 0.32466414, 0.91222045]])
>>> x.flags
  C_CONTIGUOUS : False
  F_CONTIGUOUS : True
  OWNDATA : True
  WRITEABLE : True
  ALIGNED : True
  WRITEBACKIFCOPY : False
  UPDATEIFCOPY : False
>>> y = np.ndarray(x.shape, buffer=x.flatten(), order='C')
>>> y
array([[0.4374804 , 0.30758627, 0.97045243, 0.35728207, 0.95584896],
       [0.41457004, 0.82433351, 0.44260369, 0.80797501, 0.78434397],
       [0.10598124, 0.98438937, 0.13709334, 0.24600426, 0.20042625],
       [0.73949954, 0.54399577, 0.64043765, 0.52684565, 0.89393217],
       [0.61319107, 0.67340923, 0.37922999, 0.32466414, 0.91222045]])
>>> y.flags
  C_CONTIGUOUS : True
  F_CONTIGUOUS : False
  OWNDATA : False
  WRITEABLE : True
  ALIGNED : True
  WRITEBACKIFCOPY : False
  UPDATEIFCOPY : False
>>> z = np.ndarray(y.shape, buffer=y.flatten(), order='F')
>>> z
array([[0.4374804 , 0.41457004, 0.10598124, 0.73949954, 0.61319107],
       [0.30758627, 0.82433351, 0.98438937, 0.54399577, 0.67340923],
       [0.97045243, 0.44260369, 0.13709334, 0.64043765, 0.37922999],
       [0.35728207, 0.80797501, 0.24600426, 0.52684565, 0.32466414],
       [0.95584896, 0.78434397, 0.20042625, 0.89393217, 0.91222045]])
>>> z.flags
  C_CONTIGUOUS : False
  F_CONTIGUOUS : True
  OWNDATA : False
  WRITEABLE : True
  ALIGNED : True
  WRITEBACKIFCOPY : False
  UPDATEIFCOPY : False
```

Basically I created a matrix `x` that is random with `F_CONTIGUOUS`.  Then I used `np.ndarray()` with `buffer=x.flatten()` to convert to order `C_CONTIGUOUS`.  But somehow this does not copy the matrix or transpose it (which does not make sense to me---if it is `F_CONTIGUOUS` it is column-major; if it is `C_CONTIGUOUS` it is row-major; converting between the two by flattening should result in different element orderings).  But then, when I do the same thing to convert back to `F_CONTIGUOUS`, now the matrix is transposed (but not copied).

So, that's very strange to me, but it doesn't matter, since you have written a test case with `F_CONTIGUOUS` matrices, so we know it will work either way. :+1: